docs(docker): Update Docker page w/ AMD Container Runtime Toolkit (#749)
ROCm/rocm-install-on-linux: docs/how-to/docker.rst

.. meta::
   :description: Running ROCm workloads in Docker containers on AMD GPUs
   :keywords: Docker, AMD, ROCm, GPU, container, amd-container-runtime, CDI

**************************
Run ROCm Docker containers
**************************

Docker is a popular way to run ROCm workloads in a consistent, reproducible environment.

> There are two ways to grant Docker containers access to AMD GPUs:

Reviewer: ```suggestion
There are two ways to expose AMD GPUs to Docker containers:
```

* :ref:`AMD Container Toolkit <docker-with-toolkit>` — simplifies
  GPU access within Docker environments, enhances device discovery, and enables
  better integration with modern container technologies.

* :ref:`Plain Docker device passthrough <docker-plain>` — passes GPU device nodes

>   directly to the container with ``--device``. No toolkit required.

Reviewer: ```suggestion
* :ref:`Manual Docker device passthrough <docker-plain>` — passes GPU device nodes
  directly to the container with ``--device``. No toolkit required.
```

Prerequisites
=============

Regardless of which approach you use, the following are required on the host system:

> * The AMD GPU kernel-mode driver (``amdgpu-dkms``) must be installed.

Reviewer: ```suggestion
* The AMD GPU kernel-mode driver (``amdgpu-dkms``).
```

  See `Install amdgpu-dkms
  <https://instinct.docs.amd.com/projects/amdgpu-docs/en/latest/install/package-manager-index.html>`__.

> * `Docker Engine <https://docs.docker.com/engine/install/>`__ must be installed.

Reviewer: ```suggestion
* `Docker Engine <https://docs.docker.com/engine/install/>`__.
```

.. _docker-with-toolkit:

With the AMD Container Toolkit
==============================

The `AMD Container Toolkit <https://instinct.docs.amd.com/projects/container-toolkit/en/latest/container-runtime/overview.html>`__
provides two mechanisms for GPU injection: CDI (recommended) and the
amd-container-runtime. Both require installing the toolkit first.

1. Install the toolkit by following the `Quick Start Guide (AMD Container Runtime Toolkit docs)
   <https://instinct.docs.amd.com/projects/container-toolkit/en/latest/container-runtime/quick-start-guide.html>`__.

2. See `Running Workloads (AMD Container Runtime Toolkit docs)
   <https://instinct.docs.amd.com/projects/container-toolkit/en/latest/container-runtime/running-workloads.html>`__
   to get started running ROCm applications.

.. _docker-plain:

> Without the AMD Container Toolkit (plain Docker)

Reviewer: ```suggestion
Without the AMD Container Toolkit
```
This looks too verbose. The next paragraph already explains that it only requires docker.

We can similarly avoid the word "plain" elsewhere.

Author: Makes sense. Thanks.

================================================

If you prefer not to install the AMD Container Toolkit, or you're using older
versions of Docker Engine, you can pass GPU device nodes directly to the
container using Docker's ``--device`` flag. This approach requires no
additional software beyond Docker.

.. code-block:: shell

   docker run -it --rm \
       --device /dev/kfd \
       --device /dev/dri \
       --security-opt seccomp=unconfined \
       <image>

The purpose of each option:

* ``--device /dev/kfd``

  The main compute interface, shared by all GPUs.

* ``--device /dev/dri``

  Contains the Direct Rendering Interface (DRI) device nodes for each GPU.
  Passing the whole directory grants access to all GPUs. To restrict access
  to specific GPUs, see :ref:`docker-restrict-gpus`.

* ``--security-opt seccomp=unconfined`` (optional)

  Enables memory mapping. Recommended for HPC workloads that use ``numactl``
  for GPU/CPU affinity mappings.
  See `Docker security options
  <https://docs.docker.com/reference/cli/docker/container/run/#security-opt>`_.

.. _docker-restrict-gpus:

> Restricting GPU access

Reviewer: This section could have a _tip_ at the end saying something like "container toolkit simplifies GPU selection removing the need to manually map the render nodes, we recommend to use it if your use case needs fine-grained GPU selection".
